lib/galaxy/webapps/galaxy/controllers/workflow.py

```diff
@@ -432,15 +428,29 @@ def save_workflow_as(
                     workflow_data,
                     workflow_update_options,
                 )
+            except exceptions.DuplicatedIdentifierException as e:
+                return self.message_exception(trans, e.err_msg, False)
```

Comment:
This is only necessary because you're using the controller method, which you don't need to use. The API handles this correctly, `DuplicatedIdentifierException` derives from MessageException.

Reply:
Ah ok, now I understand why the exception wasn't being caught at all